src/ledctl/ledctl.c

```diff
@@ -555,73 +779,109 @@ static led_status_t _cmdline_ibpi_parse(int argc, char *argv[])
 static led_status_t _cmdline_parse_non_root(int argc, char *argv[])
 {
 	int opt_index, opt = -1;
-	led_status_t status = LED_STATUS_SUCCESS;
+	char *shortopts;
+	struct option *longopts;
 
-	do {
-		opt = getopt_long(argc, argv, shortopt, longopt, &opt_index);
-		switch (opt) {
-		case 'v':
+	optind = test_params ? optind : 1;
```

Comment:
IMO bug prone recurrency.
I don't see we need test flag here. It has nothing to do with non_root. If non root option is detected we are printing and exiting.